### ISSUE TITLE
Respect Bullets/Numbering in non-Markdown files.

### ISSUE THREAD

**Issue by abrightwell** [dnut/Rewrap#2]
This was an [issue](https://github.com/stkb/Rewrap/issues/268) that was raised previously in the upstream Rewrap. Would be cool to see it added in future versions if this fork will become the future of the extension.

I'm happy to help with a PR, though I'm not terribly familiar with the VSCode extension workflow/framework. Regardless, I figured I'd post here just in case it were something that were 'simple' for someone else more familiar to knock out. 😁

**Comment by wycats** [comment]
Properly respecting Markdown and JSDoc would make such a huge difference to the usability of this extension.

It would make `rewrap.onSave` and `rewrap.wholeComment` much more viable for me in general.

**Comment by abrightwell** [comment]
@wycats I've started poking at it to see if I can get a PR up. Seems like there are a lot of other updates necessary to get it the extension up-to-date from tool set perspective as well. I've got a lot of updates related to that as well to bring it 'current' that I hope to PR in the near future. Hopefully that will help move this one forward as well.

**Comment by dnut** [comment]
Weird. I'm not observing these problems. I'm accustomed to markdown bullets being handled correctly. Actually the docs home page highlights this as one of the features of the extension. If it's not working, that sounds like a bug.

Regarding the link you shared https://github.com/stkb/Rewrap/issues/268, that sounds like a related but different issue. I think they were asking for the parser to be generalized and usable in other contexts, not reporting the bug that you're observing.

When I rewrap this comment, nothing changes:
```
// * A really long bullet item.
// * Another really long bullet item.
```

Then I made the comments actually long:
```
// * A really long bullet item. A really long bullet item. A really long bullet item. A really long bullet item.
// * Another really long bullet item. Another really long bullet item. Another really long bullet item.
```

After rewrap, it looks like this:

```
// * A really long bullet item. A really long bullet item. A really long bullet
//   item. A really long bullet item.
// * Another really long bullet item. Another really long bullet item. Another
//   really long bullet item.
```

I wonder what the difference is between our setups. What versions are you using (extension + vscode)? What programming language are you working with? Can you share an entire source file that reproduces the issue?

**Comment by abrightwell** [comment]
> Weird. I'm not observing these problems. I'm accustomed to markdown bullets being handled correctly. Actually the docs home page highlights this as one of the features of the extension. If it's not working, that sounds like a bug.

Yeah, I believe it behaves correctly in some file types but not all. For instance, I'm seeing it Golang files `.go` primarily.

> Regarding the link you shared [stkb#268](https://github.com/stkb/Rewrap/issues/268), that sounds like a related but different issue. I think they were asking for the parser to be generalized and usable in other contexts, not reporting the bug that you're observing.

I think this bug was the closest I could find that was potentially related. But, overall, I was thinking a more generalized approach would be what I would hope for here maybe? Unfortunately, I'm not familiar enough with the code base just yet to speak intelligently to it though. So, I'll defer to you on this one. 😁 

> When I rewrap this comment, nothing changes:
> 
> ```
> // * A really long bullet item.
> // * Another really long bullet item.
> ```
> 
> Then I made the comments actually long:
> 
> ```
> // * A really long bullet item. A really long bullet item. A really long bullet item. A really long bullet item.
> // * Another really long bullet item. Another really long bullet item. Another really long bullet item.
> ```

I just attempted in my env in a golang file and got the following:

```
// * A really long bullet item. A really long bullet item. A really long bullet
// item. A really long bullet item. * Another really long bullet item. Another
// really long bullet item. Another really long bullet item.
```

> After rewrap, it looks like this:
> 
> ```
> // * A really long bullet item. A really long bullet item. A really long bullet
> //   item. A really long bullet item.
> // * Another really long bullet item. Another really long bullet item. Another
> //   really long bullet item.
> ```

This is exactly what I was hoping for as a result. 😄 
 
> I wonder what the difference is between our setups. What versions are you using (extension + vscode)? What programming language are you working with? Can you share an entire source file that reproduces the issue?

Rewrap: dnut.rewrap-revived - Version 1.16.3
VSCode: Version: 1.96.2 (Universal)
Golang: golang.go - Version 0.44.0

FWIW, I disabled the golang extension to see if that was possibly interfering, but it didn't have any impact.

Here's a sample file though: https://gist.github.com/abrightwell/bac0abab3c5d3ce8c51534b5beb5152b

**Comment by dnut** [comment]
Thanks for the additional context. I have reproduced this in go files.

Go comments are parsed as godocs instead of markdown, which seems to be the cause.

I'm able to get markdown working in go files if I sacrifice the custom logic for parsing godocs and fall back to the standard markdown parser. I don't think this is an acceptable solution, but here's what it looks like:

```diff
diff --git a/core/Parsing.Documents.fs b/core/Parsing.Documents.fs
index 6380590..5aa4a40 100644
--- a/core/Parsing.Documents.fs
+++ b/core/Parsing.Documents.fs
@@ -117,7 +117,7 @@ let mutable languages = [
             ]
         )
     lang "FIDL" "" ".fidl" <| sc [line "///?"]
-    lang "Go" "" ".go" <| sc [block' ("", "") (@"/\*", @"\*/") godoc; line' "//" godoc]
+    lang "Go" "" ".go" <| sc [block' ("", "") (@"/\*", @"\*/") markdown; line' "//" markdown]
     lang "Git commit" "git-commit" "tag_editmsg" <| docOf markdown
     lang "GraphQL" "" ".graphql|.gql" <| sc [line "#"; block (@".*?""""""", "\"\"\"")]
     lang "Groovy" "" ".groovy" java
```

Ideally there would be a solution that could unify the markdown parser with the godoc parser.
